### PR TITLE
Handle special characters in snowflake query

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Related to https://flyte-org.slack.com/archives/CP2HDHKE1/p1664792800827789?thread_ts=1663843915.889099&cid=CP2HDHKE1.

[snowflake docs](https://docs.snowflake.com/en/sql-reference/data-types-text.html#dollar-quoted-string-constants)

Replace "\n" to "\\n" to fix the 400 error.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/core/base_sql_task.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def __init__(
             task_config=task_config,
             **kwargs,
         )
-        self._query_template = query_template
+        self._query_template = query_template.replace("\n", "\\n").replace("\t", "\\t")
```
Comment (eapolinario) [inline flytekit/core/base_sql_task.py:44]: are there any other characters or escape sequences that are specific to snowflake? I tried to read https://docs.snowflake.com/en/sql-reference/data-types-text.html, but I'm still not sure if we have to handle some of those.

Comment (pingsutw) [reply flytekit/core/base_sql_task.py:44]: I feel like some of those can handle by users themself. if we triple quote like
```
 query = """select 1
 from table"""
```
python will convert it to `select 1\nfrom table`, so that's why we help them handle this special character in flytekit.

However, if they want to run something like `select $1, $2 from ('Backslash','C:\user');`, they are able to explicit convert it to `select $1, $2 from ('Backslash','C:\\user');` by themself to make it work in snowflake.

**Review by eapolinario — APPROVED** [review]
